parser/parser.go

```diff
@@ -33,173 +33,151 @@ var (
-// it initializes the failure log file for subsequent lines.
-func (fp *FuzzProcessor) handleFailureDetection(line string) error {
-	// Check if the line contains the failure marker.
-	if strings.Contains(line, "--- FAIL:") {
-		// Mark that a failure has been detected.
-		fp.State.SeenFailure = true
-
-		// Construct the log file name and path for storing failure
-		// details.
-		logFileName := fmt.Sprintf("%s_failure.log", fp.target)
-		logPath := filepath.Join(fp.cfg.FuzzResultsPath, logFileName)
-
-		// Ensure the FuzzResultsPath directory exists (creates parents
-		// as needed)
-		if err := utils.EnsureDirExists(
-			fp.cfg.FuzzResultsPath); err != nil {
-			return fmt.Errorf("failed to create fuzz result file: "+
-				"%w", err)
+	// Ensure the log file is closed at the end.
+	defer func() {
+		if err := fp.logFile.Close(); err != nil {
+			fp.logger.Error("Failed to close log file", "error",
+				err)
 		}
+	}()
+
+	fp.logger.Info("Failure log initialized", "path", logPath)
 
-		// Initialize the log writer with the constructed path.
-		if err := fp.logWriter.Initialize(logPath); err != nil {
-			return fmt.Errorf("log writer initialization failed: "+
-				"%w", err)
+	var errorData string
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		fp.logger.Info("Fuzzer output", "message", line)
+
+		// Write the current line to the failure log file.
+		_, err = fp.logFile.WriteString(line + "\n")
+		if err != nil {
+			fp.logger.Error("Failed to write log line", "error",
+				err)
+			return
 		}
 
-		fp.logger.Info("Failure log initialized", "path", logPath)
-	}
-	return nil
-}
+		// If error data has already been captured, skip further
+		// extraction.
+		if errorData != "" {
+			continue
+		}
 
-// handleFailureLine writes the line to the log, then on the first occurrence
-// of a failure-input marker extracts the testcase and reads its contents.
-func (fp *FuzzProcessor) handleFailureLine(line string) error {
-	// Log the current line to the failure log file.
-	if err := fp.logWriter.WriteLine(line); err != nil {
-		return fmt.Errorf("failed to write log line: %w", err)
-	}
+		// Parse the line to extract the fuzz target and ID (hex) of the
+		// failing input.
+		// When a fuzz target encounters a failure during f.Add, the
+		// crash is printed, but no input is saved to testdata/fuzz.
+		//
+		// The log output typically appears as:
+		//   failure while testing seed corpus entry: FuzzFoo/seed#0
+		//
+		// As a result, no error data will be printed.
+		target, id := parseFailureLine(line)
+		// If either target or ID is empty, skip further processing.
+		if target == "" || id == "" {
+			continue
+		}
 
-	// If the failing input has already been printed, no further action is
-	// needed.
-	if fp.State.InputPrinted {
-		return nil
+		// Read and store the input data associated with the failing
+		// target and ID.
+		errorData = fp.readFailingInput(target, id)
```

Comment:
Nit:  we can use a more descriptive variable name

```suggestion
		failingInputString = fp.readFailingInput(target, id)
```